tripal/src/Services/TripalPublish.php

```diff
@@ -733,25 +741,35 @@ protected function insertFieldItems($field_name, $matches, $titles, $entities, $
     $init_sql = rtrim($init_sql, ", ");
     $init_sql .= ") VALUES\n";
 
+    $i = 0;
     $j = 0;
     $total = 0;
     $batch_num = 1;
     $sql = '';
     $args = [];
+    $num_inserted = 0;
 
-    // Iterate through the matches.
+    // Iterate through the matches. Each match corresponds to a single
+    // entity. The titles provided should be in order of the entities
+    // in the matches array.
     foreach ($matches as $match) {
-      $title = $titles[$total];
+
+      $title = $titles[$i];
       $entity_id = $entities[$title];
+      $i++;
 
+      // Iterate through the "items" of each feild and insert a record value
+      // for each item.
       $num_delta = count(array_keys($match[$field_name]));
       for ($delta = 0; $delta < $num_delta; $delta++) {
         $j++;
         $total++;
 
         // No need to add items to those that are already published.
         if (array_key_exists($entity_id, $existing)) {
-          continue;
+          if (array_key_exists($delta, $existing[$entity_id])) {
+            continue;
+          }
         }
```

Comment:
I can't do a suggestion here, but the unpublished fields happens because of the continue. If we happen to be on the LAST iteration, we never get to the code below to perform the sql query!
We need to change this to something like
```
if (!array_key_exists($entity_id, $existing) or !array_key_exists($delta, $existing[$entity_id])) {
  then do the sql code below up to and including $num_inserted++;
}
```

Reply:
I've integrated this fix. Thanks for checking deep enough to spot this.